presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -478,14 +496,17 @@ public boolean hasNext()
                 return output;
             }
 
-            //  TODO: Implement task stats collection
-            //  TaskStats taskStats = taskContext.getTaskStats();
-            //  byte[] taskStatsSerialized = taskInfoJsonCodec.toJsonBytes(taskStats);
-            //  taskStatsCollector.add(new SerializedTaskStats(taskStatsSerialized));
-
             // task finished
             TaskState taskState = taskStateMachine.getState();
             checkState(taskState.isDone(), "task is expected to be done");
+
+            TaskInfo taskInfo = createTaskInfo(taskContext, taskStateMachine, taskInstanceId, outputBuffer);
+            SerializedTaskInfo serializedTaskInfo = new SerializedTaskInfo(
+                    taskInfo.getTaskId().getStageExecutionId().getStageId().getId(),
+                    taskInfo.getTaskId().getId(),
+                    compress(taskInfoJsonCodec.toJsonBytes(taskInfo)));
+            taskInfoCollector.add(serializedTaskInfo);
```

Comment:
curious: how do we avoid `byte[]` being displayed on Spark UI?

Reply:
Answered in the previous comment